src/Illuminate/Container/Container.php

```diff
@@ -401,6 +414,28 @@ public function extend($abstract, Closure $closure)
         }
     }
 
+    /**
+     * Add a global extender callback to the container.
+     *
+     * @param  \Closure  $closure
+     * @return void
+     *
+     * @throws \InvalidArgumentException
+     */
+    protected function addGlobalExtender(Closure $closure)
+    {
+        foreach ($this->instances as $abstract => $instance) {
+            $this->instances[$abstract] = $closure($instance, $this);
+            $this->rebound($abstract);
+        }
+
+        foreach (array_keys($this->resolved) as $abstract) {
+            $this->rebound($abstract);
```

Comment:
This feels a bit odd. We fire the rebound callback for everything every resolved by the container, even if it was not affected by this global extender?

Reply:
I agree. At first I planned on firing the rebound callback only for resolved instances but then I saw that [we currently also fire them for resolved bindings](https://github.com/laravel/framework/pull/35202/files#diff-cc160328d7d447677b8d359453f4416c820a9117c5efd500a3f0361d1d5bab1aR409-R413) so I decided to stay consistent with the current behaviour.

That being said, I understand that with specific extenders, we can be pretty sure the resolved bindings are affected by the extender whereas we cannot make the same assumption with global extenders.

I think a good compromise would be to:
- Keep firing the rebinding callbacks for resolved instances — since they are re-computed by the global extender.
- Stop firing the rebinding callbacks for resolved bindings — since they are not being re-computed and we cannot assume that the global extender affects them.

What do you think?